src/searchbar.cpp

```diff
@@ -263,6 +270,9 @@ void SearchBarLineEdit::fetchSuggestions(NewSuggestionHandlerFuncPtr callback)
                 }
 
                 m_suggestionModel.append(suggestionList);
+                bool hasFullText = suggestionList.size() > 0 && suggestionList.back().isFullTextSearchSuggestion();
+                int trueFetchSize = suggestionList.size() - (hasFullText ? 1 : 0);
+                m_moreSuggestionsAreAvailable = trueFetchSize >= SuggestionListWorker::getFetchSize();
```

Comment:
Risk of underflow - when `suggestionList.size() == 0` the initializer expression of `trueFetchSize` is questionable if the type of `suggestionList.size()` is unsigned. The following should be safer:

```c++
m_moreSuggestionsAreAvailable = suggestionList.size() >= SuggestionListWorker::getFetchSize() + hasFullText;
```

Reply:
Done

Reply:
After going to bed I figured out that here there was no risk of underflow because `hasFullText` is `false` when `suggestionList.size() == 0` and wanted to delete this comment in the morning. But the general advice still holds - when operating in the vicinity of `0`, addition is safer than subtraction. Similarly, when operating in the vicinity of the upper limit of an integer numeric type, subtraction is safer than addition.